### PR TITLE
4.8 tasks list page maslova

### PR DESCRIPTION
**Pull request by maslovakat**

### CODE REVIEW

File: front/src/containers/list-of-group/index.js [hunk 1]
```diff
@@ -28,8 +28,9 @@ class ListOfGroup extends Component {
 
     componentDidMount() {
         const token = localStorage.getItem('token');
-        const userId = JSON.parse(localStorage.getItem('user')).id;
-        const url = `${process.env.REACT_APP_GROUPS_BY_MENTOR}${userId}`;
+        // const userId = JSON.parse(localStorage.getItem('user')).id;
+        const url = `${process.env.REACT_APP_GROUPS_BY_MENTOR}`;
+        console.log(url)
```
Comment (Alan18081) [inline front/src/containers/list-of-group/index.js]: Useless `console.log`

File: front/src/containers/list-of-group/index.js [hunk 2]
```diff
@@ -28,8 +28,9 @@ class ListOfGroup extends Component {
...
         const headers = { headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}`, } };
 
         axios.get(url, headers)
```
Comment (Alan18081) [inline front/src/containers/list-of-group/index.js]: Please, move request to the saga

File: front/src/pages/tasks-list/index.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+const { TextArea } = Input;
+const layout = {
+    labelCol: { span: 8 },
+    wrapperCol: { span: 16 }
+};
+
+class TasksList extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
```
Comment (Alan18081) [inline front/src/pages/tasks-list/index.js:32]: Define separated reducer for tasks

File: front/src/containers/list-of-group/index.js (hunk 1 above)
Comment (maslovakat) [reply front/src/containers/list-of-group/index.js]: fixed

File: front/src/containers/list-of-group/index.js (hunk 2 above)
Comment (maslovakat) [reply front/src/containers/list-of-group/index.js]: done

File: front/src/pages/tasks-list/index.js (hunk 3 above)
Comment (maslovakat) [reply front/src/pages/tasks-list/index.js:32]: done

File: front/src/common/redux/tasks/task.saga.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        const response = yield call(changeTask, payload);
+
+        yield put({ type: types.CHANGE_TASK_SUCCESS, response });
+    } catch (error) {
+        yield put({ type: types.CHANGE_TASK_ERROR, error });
+    }
+}
+
+function* watchTasks() {
+    yield takeEvery(types.GET_ALL_TASKS, getTasksSaga);
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.saga.js]: Please, move all `takeEvery` to `tasksSaga` and remove `watchTasks`

File: front/src/common/redux/tasks/task.api.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import axios from 'axios';
+
+async function getTasks(payload) {
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Pass actual data, not the whole action object

File: front/src/common/redux/tasks/task.api.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    const { page, groupId } = payload.data;
+    const token = localStorage.getItem('token');
+    const url = `${process.env.REACT_APP_TASKS}?page=${page}&limit=10&groupId=${groupId}`;
+    const headers = { headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}`, } };
+
+    return axios.get(url, headers)
+        .then(response => response)
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Please, return only actual data, not the whole http response object

File: front/src/common/redux/tasks/task.api.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+async function deleteTask(payload) {
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Pass actual data, not the whole action object

File: front/src/common/redux/tasks/task.api.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    const { id } = payload;
+    const token = localStorage.getItem('token');
+    const url = `${process.env.REACT_APP_TASKS}/${id}`;
+    const headers = { headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}`, } };
+
+    return axios.delete(url, headers)
+        .then(response => response);
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Please, return only actual data, not the whole http response object

File: front/src/common/redux/tasks/task.api.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+
+async function changeTask(payload) {
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Pass actual data, not the whole action object

File: front/src/common/redux/tasks/task.api.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+    const { id, field } = payload.data;
+    const token = localStorage.getItem('token');
+    const url = `${process.env.REACT_APP_TASKS}/${id}`;
+    const headers = { headers: { 'Content-Type': 'application/json', 'Authorization': `Bearer ${token}`, } };
+
+    return axios.put(url, JSON.stringify(field), headers)
+        .then(response => response)
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.api.js]: Please, return only actual data, not the whole http response object

File: front/src/common/redux/tasks/task.saga.js [hunk 11]
```diff
@@ -0,0 +1,45 @@
+import { put, call, all, takeEvery } from "redux-saga/effects";
+import { getTasks, deleteTask, changeTask } from "./task.api";
+import * as types from "./types";
+
+function* getTasksSaga(payload) {
+    try {
+        const response = yield call(getTasks, payload);
+
+        yield put({ type: types.GET_ALL_TASKS_SUCCESS, response });
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.saga.js]: Please, provide key `payload` for the data of action

File: front/src/common/redux/tasks/task.saga.js [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+    } catch (error) {
+        yield put({ type: types.GET_ALL_TASKS_ERROR, error });
+    }
+}
+
+function* deleteTaskSaga(payload) {
+    try {
+        const response = yield call(deleteTask, payload);
+
+        yield put({ type: types.DELETE_TASK_SUCCESS, response });
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.saga.js]: Please, provide key `payload` for the data of action

File: front/src/common/redux/tasks/task.saga.js [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    } catch (error) {
+        yield put({ type: types.DELETE_TASK_ERROR, error });
+    }
+}
+
+function* changeTaskSaga(payload) {
+    try {
+        const response = yield call(changeTask, payload);
+
+        yield put({ type: types.CHANGE_TASK_SUCCESS, response });
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.saga.js]: Please, provide key `payload` for the data of action

File: front/src/common/redux/tasks/task.saga.js (hunk 4 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.saga.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 5 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 6 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 7 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 8 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 9 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.api.js (hunk 10 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.api.js]: done

File: front/src/common/redux/tasks/task.saga.js (hunk 11 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.saga.js]: done

File: front/src/common/redux/tasks/task.saga.js (hunk 12 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.saga.js]: done

File: front/src/common/redux/tasks/task.saga.js (hunk 13 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.saga.js]: done

File: front/src/common/redux/tasks/task.reducer.js [hunk 14]
```diff
@@ -8,13 +8,13 @@ const tasksState = {
 }
 
 export function tasks(state = tasksState, action) {
-    const { response } = action;
-
+    console.log(action)
```
Comment (Alan18081) [inline front/src/common/redux/tasks/task.reducer.js]: Useless `console.log`

**Review by Alan18081 — APPROVED** [review]

File: front/src/common/redux/tasks/task.reducer.js (hunk 14 above)
Comment (maslovakat) [reply front/src/common/redux/tasks/task.reducer.js]: done

**Pull request merged**